assets/js/components/ErrorNotice.test.js

```diff
@@ -0,0 +1,243 @@
+/**
+ * ErrorNotice component tests.
+ *
+ * Site Kit by Google, Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Internal dependencies
+ */
+import { createTestRegistry, provideModules } from '../../../tests/js/utils';
+import {
+	ERROR_CODE_MISSING_REQUIRED_SCOPE,
+	ERROR_REASON_INSUFFICIENT_PERMISSIONS,
+} from '../util/errors';
+import { fireEvent, render } from '../../../tests/js/test-utils';
+import ErrorNotice from './ErrorNotice';
+import { MODULES_ANALYTICS } from '../modules/analytics/datastore/constants';
+
+describe( 'ErrorNotice', () => {
+	let registry;
+	let invalidateResolutionSpy;
+	const moduleName = 'test-module';
+
+	beforeEach( () => {
+		registry = createTestRegistry();
+		provideModules( registry, [
+			{ slug: moduleName, name: 'Test Module' },
+		] );
+		invalidateResolutionSpy = jest.spyOn(
+			registry.dispatch( MODULES_ANALYTICS ),
+			'invalidateResolution'
+		);
+	} );
+
+	afterEach( () => invalidateResolutionSpy.mockReset() );
+
+	it( "should not render the `Retry` button if the error's `selectorData.name` is not `getReport`", () => {
+		const { queryByText } = render(
+			<ErrorNotice
+				error={ {
+					code: 'test-error-code',
+					message: 'Test error message',
+					data: {
+						reason: ERROR_REASON_INSUFFICIENT_PERMISSIONS,
+					},
+					selectorData: {
+						args: [],
+						name: 'getAccountID',
+						storeName: MODULES_ANALYTICS,
+					},
+				} }
+			/>,
+			{
+				registry,
+			}
+		);
+
+		expect( queryByText( /retry/i ) ).not.toBeInTheDocument();
+	} );
```

Comment:
This test isn't actually accurate, and is failing because of the `reason: ERROR_REASON_INSUFFICIENT_PERMISSIONS` in the error. If you change `reason: ''`, this test will fail.

That's because the `err.selectorData?.name === 'getReport' &&` condition wasn't transferred over to the new `isErrorRetryable` function.

```suggestion
	it( "should not render the `Retry` button if the error's `selectorData.name` is not `getReport`", () => {
		const { queryByText } = render(
			<ErrorNotice
				error={ {
					code: 'test-error-code',
					message: 'Test error message',
					data: {
						reason: '',
					},
					selectorData: {
						args: [],
						name: 'getAccountID',
						storeName: MODULES_ANALYTICS,
					},
				} }
			/>,
			{
				registry,
			}
		);

		expect( queryByText( /retry/i ) ).not.toBeInTheDocument();
	} );
```